examples/AvroBlogExamples/Program.cs

```diff
@@ -35,13 +35,10 @@ class Program
     {
         static void ProduceGeneric(string bootstrapServers, string schemaRegistryUrl)
         {
-            var config = new Dictionary<string, object>
-            {
-                { "bootstrap.servers", bootstrapServers },
-                { "schema.registry.url", schemaRegistryUrl }
-            };
+            var producerConfig = new ProducerConfig { BootstrapServers = bootstrapServers };
 
-            using (var producer = new Producer<Null, GenericRecord>(config, null, new AvroSerializer<GenericRecord>()))
+            using (var serdeProvider = new AvroSerdeProvider(new AvroSerdeProviderConfig { SchemaRegistryUrl = schemaRegistryUrl }))
+            using (var producer = new Producer<Null, GenericRecord>(producerConfig, null, serdeProvider.CreateValueSerializer<GenericRecord>()))

```

Comment:
it's not great that the type system isn't helping ensure `CreateValueSerializer` isn't being used to create a key serializer. also, there's no run time check of this currently. doing something better here is a todo before 1.0. the 'provider' idea is much better than having the serdes IDisposable though, and broadly speaking the way forward.